
<!-- Copyright (C) 2019-2021 LEIDOS.
Licensed under the Apache License, Version 2.0 (the "License"); you may not
use this file except in compliance with the License. You may obtain a copy of
the License at
http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
License for the specific language governing permissions and limitations under
the License.
-->

<?xml version="1.0"?>
<launch>
    <arg name="vehicle_calibration_dir" default="/opt/carma/vehicle/calibration"/>
    <!-- Pure Pursuit Node -->
    <group>
        <node pkg="pure_pursuit" type="pure_pursuit" name="pure_pursuit" output="log">
            <remap from="final_waypoints" to="carma_final_waypoints"/>
            <param name="is_linear_interpolation" value="True"/>
            <param name="publishes_for_steering_robot" value="True"/>
            <rosparam command="load" file="$(arg vehicle_calibration_dir)/pure_pursuit/calibration.yaml"/>
        </node>
    </group>
    <!-- Pure Pursuit Wrapper Node -->
    <group>
       <remap from="final_waypoints" to="carma_final_waypoints"/>
        <node pkg="pure_pursuit_wrapper" type="pure_pursuit_wrapper_node" name="pure_pursuit_wrapper_node">
            <rosparam command="load" file="$(find pure_pursuit_wrapper)/config/default.yaml" />
        </node>
    </group>
</launch>
